lanelet2/lanelet2_routing/src/Route.cpp

```diff
@@ -417,5 +428,22 @@ Route::Errors Route::checkValidity(bool throwOnError) const {
   return errors;
 }
 
+void Route::setEndPoint(lanelet::Point3d end_point)
+{
+  if (!boost::geometry::covered_by(end_point.basicPoint2d(), shortestPath_.back().polygon2d()))
+  {
+    std::cerr << "Given end_point is not inside the last lanelet of the route" << std::endl;
```

Comment:
Lets have this throw an exception or return an error code.

Reply:
throwing an error